### DISCUSSION TITLE
Integrating Scheme code in a .NET HTTP application

### DISCUSSION THREAD

**Discussion by bcalco** [leppie/IronScheme#115] [Discussions]
@Leppie On the theory that you have meditated on this topic longer than I have, I am seeking suggestions for approaching the task of integrating IronScheme code in the context of lower level HttpServer applications. The whole async/await thing, plus executing handlers that call into IronScheme code ,and defining routes.

As I progress in my exploration of this issue, I'll post things I learn here; but I am also eager to save myself time where known dead-ends exist, and as they say, you don't know know what you don't know. 

I am aware of some of the web related libraries in `(ironscheme web ...)` but confess to needing some explanation and examples of using the abstractions (especially the macros) found there. The "running the webserver" wiki entry is not relevant to my goals, and feels dated. My goals are twofold:

1. Expose IronScheme code to handlers from the host, such that the host can call into Scheme handlers and to a degree even hand off to Scheme code setting of routes;
2. Create a web server library in 100% IronScheme so users can easily spin up Node.js Express-style one-off web servers.

And of course, I want to deal with the possibly different low level web server alternatives on both Standard and Core. (That is, the Scheme code abstracts away the low-level particulars of the diverse .NET runtimes, as necessary.)

**Comment by leppie** [comment]
I would really have to look at this as I am not really familiar latest ASP.NET Core way of doing these things.

https://github.com/IronScheme/TwitterWebHook is the quick and dirty way.
